nano/lib/numbers.cpp

```diff
@@ -61,6 +62,16 @@ std::string nano::public_key::to_account () const
 	return result;
 }
 
+nano::public_key::public_key () :
+	uint256_union{ 0 }
+{
+}
+
+const nano::public_key & nano::public_key::null () const
+{
+	return nano::hardened_constants::get ().not_an_account;
```

Comment:
I expected this to return a reference to a static public_key rather construct a temporary public key.

Reply:
It is not constructing a temporary key. It is returning a lazily initialized reference.

Reply:
But, will it return the same exact object every time?
It would be great if there was one and only one null object.

Reply:
Yeah, that's exactly the way it behaves.